Issue #1945 fix: Minimized copied code usage by common function for v…

Yadnyawalkya opened the pull request:
This PR is a fix for [#1945](https://github.com/fedora-infra/bodhi/issues/1945) issue.

> There are several validators that make sure that various kwargs reference existing username. For example, check out validate_username(), validate_update_owner(), and validate_ignore_user(). These validators are nearly identical to the point where it'd be easy to write a common method they all use, or a validator generator function.

Copied code of validate_username(), validate_update_owner(), and validate_ignore_user() functions can be minimized with having common function for all. Please feel free to add comment if there are any validation functions are left or not covered in this PR. 

Peace out !✌

bowlofeggs commented on bodhi/server/validators.py:
```diff
@@ -800,48 +778,60 @@ def validate_update(request, **kwargs):
 
 
-def validate_update_owner(request, **kwargs):
+def common_validate(param, request):
```
Let's give this a more descriptive name, perhaps something like "```ensure_user_exists()```.

Yadnyawalkya replied:
Nice, make sense!

bowlofeggs commented on bodhi/server/validators.py:
```diff
@@ -800,48 +778,60 @@ def validate_update(request, **kwargs):
+def common_validate(param, request):
     """
-    Make sure the referenced update owner is an existing user.
+    Common fuction to be called for validatation.
```
Let's specify here that this is specifically for ensuring the referenced user exists. Perhaps we could say "Ensure the user referenced by param exists, and if it does replace it with the User object."""

bowlofeggs commented on bodhi/server/validators.py:
```diff
@@ -800,48 +778,60 @@ def validate_update(request, **kwargs):
+    Common fuction to be called for validatation.
 
     Args:
+        param (string): Variable to pass as parameter.
```
Let's make this a little more specific too - perhaps "Request parameter that references a username to be validated."""

bowlofeggs reviewed (changes requested):
Looks nice! Just a few documentation fixes and a function rename, and we'll be good to merge. Thanks!

bowlofeggs commented:
jenkies test

bowlofeggs commented:
Please also adjust your commit message to have this on its own line after the commit message body:

```
fixes #1945
```

That will cause GitHub to close #1945 automatically when this is merged.

Yadnyawalkya reviewed (commented):
Thanks for review! :) 👍  
@bowlofeggs, please have a look at recent changes.
Let me know if anything required !

Commit:
fixes #1945: Common function for validators

Copied code of validate_username(), validate_update_owner(), and
validate_ignore_user() functions can be minimized with having
common function for all.Feel free to add more validation functions
if any left or not covered in this PR. (Updated with requested changes)

fixes #1945

Signed-off-by: Yadnyawalk Tale <ytale@redhat.com>

bowlofeggs reviewed (approved):
Looks good to me! The CI system will likely not pass this PR today because Fedora is moving its servers and the container registry was not available earlier today. If the tests don't pass I'll retry them again soon. Once they pass, we are good to merge. Thanks for the patch!

bowlofeggs commented:
jenkies test

bowlofeggs commented:
jenkies test

bowlofeggs commented on bodhi/server/validators.py:
```diff
@@ -800,48 +778,60 @@ def validate_update(request, **kwargs):
+def ensure_user_exists(param, request):
     """
-    Make sure the referenced update owner is an existing user.
+    Ensure the user referenced by param exists and if it does replace it with the User object
```
The test failure wants you to end this sentence with a period:

```
bodhi/server/validators.py:807 in public function `ensure_user_exists`:
        D400: First line should end with a period (not 't')
```

You can manually run pydocstyle on the code to make sure it passes with tox, or by running this:

```
$ pydocstyle  --add-ignore D413 bodhi/server/validators.py
```

Yadnyawalkya replied:
@bowlofeggs  Added period.

bowlofeggs commented:
@Yadnyawalkya do you still want to work on this?

Commit:
fixes #1945: Added changes

Copied code of validate_username(), validate_update_owner(), and
validate_ignore_user() functions can be minimized with having
common function for all.Feel free to add more validation functions
if any left or not covered in this PR. (Updated with requested changes)

fixes #1945

Signed-off-by: Yadnyawalkya <ytale@redhat.com>

Yadnyawalkya commented:
Hey @bowlofeggs , Thanks you very much for kind reminder, I turned off my notifications and forgot to update this changes! 

Just updated that, surely like to contribute more on this side! Best regards!:)

bowlofeggs commented:
jenkies test

bowlofeggs reviewed (approved):
Thanks!

bowlofeggs commented:
This patch has been included in the [```3.3.0b0``` build](https://bodhi.stg.fedoraproject.org/docs/user/release_notes.html), which is currently deployed to [staging](https://bodhi.stg.fedoraproject.org).
